packages/realm-react/README.md

```diff
@@ -46,69 +44,170 @@ class Task extends Realm.Object {
+
+To get started with `@realm/react`, one must create a context object with `createRealmContext`.  The context object will contain a Realm Context Provider, which will have an open Realm as its context, and a set of hooks that access the Realm context.
+The configuration for the Realm context can be give to `createRealmContext` or be set directly on the Realm Context Provider props.  A Realm will be opened with this combined configuration when the Realm Context Provider is rendered.  A fallback component can optionally be rendered until the Realm is opened.  This is useful for projects using Realm Sync.  Here is an example of how to setup Realm React with a Task model:
+
+```typescript
+import {createRealmContext} from '@realm/react';
+
+const myRealmConfig = { schema: [Task, User, /*...*/]};
+
+const { RealmProvider, useRealm, useObject, useQuery } = createRealmContext(myRealmConfig);
+
+const AppWrapper = () => {
   return (
     <RealmProvider>
-      <App />
-    </RealmProvider>
-  );
-};
+      <App/>
+    <RealmProvider>
+  )
+}
+
 ```
-### Implement Hooks in Child Components
 
-The hooks created by `createRealmContext` can now be used by any child component.
+#### Multiple Realms
+`createRealmContext` can be called multiple times if you app requires more than one realm.  In that case, you would have multiple `RealmProvider`s that wrap your app and must use the hooks from the context you wish to access.
 
 ```tsx
-import RealmContext from './realm';
+const { RealmProvider: PublicRealmProvider, useRealm: usePublicRealm, useObject: usePublicObject, useQuery: usePublicQuery } = createRealmContext(publicConfig);
+const { RealmProvider: PrivateRealmProvider, useRealm: usePrivateRealm, useObject: usePrivateObject, useQuery: usePrivateQuery } = createRealmContext(privateConfig);
+```
 
-const {useRealm, useQuery, useObject} = RealmContext
+It is also possible to call it without any config, in the case that you want to do all your configuration through the RealmProvider props.
+### RealmProvider
 
-function MyComponent({someId}){
-  const realm = useRealm();
-  const tasks = useQuery(Task);
-  const singleTask = useObject(Task, taskId);
+In the example above, we used the RealmProvider without any props.  It is however, possible to configured Realm through props on the RealmProvider.
 
-  // sort collection with useMemo
-  const sortedTasks = useMemo( () => tasks.sorted("createdAt"), [tasks])
+```tsx
+const AppWrapper = () => {
+  return (
+    <RealmProvider schema={[Task, User, /*...*/]} onFirstOpen={(realm) => {/*initialize realm with some data*/}}>
+      <App/>
+    <RealmProvider>
+  )
+}
+```
 
-  // make sure the data is there
-  if(!sortedTasks || !someObject){
-    return null
-  }
+The `RealmProvider` also comes with a fallback prop that can be used for sync conifigurations.  It can take time for larger datasets to sync, especially if its the first time.  In that case, it is recommended to provide a loading component as a fallback.
```

Comment:
```suggestion
The `RealmProvider` also comes with a fallback prop that can be used for sync conifigurations.  It can take time for larger datasets to sync, especially if it's the first time.  In that case, it is recommended to provide a loading component as a fallback.
```